integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,244 @@
+        log.info("GreenMail Server stopped!");
+    }
+
+    public static GreenMailUser getGreenMailUser() {
+        return greenMailUser;
+    }
+
+    /**
+     * Connecting to the pop3 mail store.
+     *
+     * @return pop3 store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getPopConnection() throws ESBMailTransportIntegrationTestException, MessagingException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName(PROTOCOL_POP3, BIND_ADDRESS, 3110, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        Store store = session.getStore(urlName);
+        store.connect();
+        return store;
+    }
+
+    /**
+     * Connecting to the imap mail store.
+     *
+     * @return imap store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getImapConnection() throws ESBMailTransportIntegrationTestException, MessagingException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName(PROTOCOL_IMAP, BIND_ADDRESS, 3143, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        Store store = session.getStore(urlName);
+        store.connect();
+        return store;
+    }
+
+    /**
+     * Check mail folder for an email using subject.
+     *
+     * @param emailSubject
+     * @param folder
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static boolean isMailReceivedBySubject(String emailSubject, String folder, String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        boolean emailReceived = false;
+        Folder mailFolder;
+        Store store;
+        if (PROTOCOL_POP3.equals(protocol)) {
+            store = getPopConnection();
+        } else {
+            store = getImapConnection();
+        }
+        try {
+            mailFolder = store.getFolder(folder);
+            mailFolder.open(Folder.READ_WRITE);
+            SearchTerm searchTerm = new AndTerm(new SubjectTerm(emailSubject), new BodyTerm(emailSubject));
+            Message[] messages = mailFolder.search(searchTerm);
+            for (Message message : messages) {
+                if (message.getSubject().contains(emailSubject)) {
+                    log.info("Found the Email with Subject : " + emailSubject);
+                    emailReceived = true;
+                    break;
+                }
+            }
+        } finally {
+            if (store != null) {
+                store.close();
+            }
+        }
+        return emailReceived;
+    }
+
+    /**
+     * Check inbox and make sure a particular email is deleted.
+     *
+     * @param emailSubject
+     * @param protocol
```

Comment:
Let's describe all parameters, return statements and exceptions in doc comments

Reply:
done